### ISSUE TITLE
Wrong syntax hilight

### ISSUE THREAD

**Issue by gtk2k** [Microsoft/vscode#16317]
- VSCode Version: 1.7.2
- OS Version: Win10 Home 14971.1000

Steps to Reproduce:

1. Language mode switch to "JavaScript"
2.  this code copy & paste 
```js
// I want syntax highlighting
r = [v, (a = value, Math_abs(a) >= 1 ? a > 0 ? (Math_min(Math_floor(a / 10), 100) | 0) : ((a - 1) / 1) : 0)], b[t] = c[0], b[t] = b[1];

// bad syntax highlighting
r = [v, (a = value, Math_abs(a) >= 1 ? a > 0 ? (Math_min(Math_floor(a / 10), 100) | 0) : ((a - 1) / +1) : 0)], b[t] = c[0], b[t] = b[1];
```

Line #2 wont work 
Line #5 diff only "+" wrong syntax highlighting

**Comment by chrmarti** [comment]
@gtk2k Could you attach a screenshot showing the problem? I cannot reproduce it in my setup.

**Comment by mjbvz** [comment]
This is fixed in the insiders builds:

<img width="780" alt="screen shot 2016-12-01 at 10 50 43 am" src="https://cloud.githubusercontent.com/assets/12821956/20807517/0b383f92-b7b4-11e6-99fc-59bdc06720f3.png">


1.7.2 is top, insiders 1.8 is bottom.


Please try out the latest insiders builds. The fix should become available in 1.8

Thanks.

**Comment by gtk2k** [comment]
@chrmarti @mjbvz thanks!
